R/outpack_store.R

```diff
@@ -32,11 +32,10 @@ file_store <- R6::R6Class(
                            paste(sprintf("  - %s", missing), collapse = "\n"))
         stop(not_found_error(message, missing))
       }
-      fs::dir_create(dirname(dst))
 
-      # Set copy_mode = FALSE: files in the store are read-only. It's easier on
-      # the user if we make them writable again.
-      copy_files(src, dst, overwrite = overwrite, copy_mode = FALSE)
+      # Files in the archive are read-only. It's easier on the user if we make
+      # them writable again.
```

Comment:
Huh my mistake here is actually setting `make_writable = FALSE` instead of `TRUE`. I'm amazed there's not a test to catch that. I'll fix that, also need to change `archive` to `store` in the comment.

But otherwise the comment is correct. This function moves files out of the store and into the user's directory (either the drafts folder if running a packet, the src folder if running interactively or anywhere if the user just called `orderly_copy_files`).

Reply:
Hah, so the tests passed because it was doing `orderly_copy_files` twice and whether that works. Turns out that now works even if the file is read-only because the new `overwrite = TRUE` behaviour will overwrite read-only files, whereas the old one did not.

I think I might actually tweak the `copy_files` function to not allow that anymore.